### PR TITLE
removed a mutex from nodes coordinator, removed some weak warnings

### PR DESCRIPTION
**Pull request by iulianpascalau**
- removed the mutex inside `epochNodesConfig` struct
- removed some weak warnings
- added TODOs to further refactor this package

### CODE REVIEW

File: sharding/indexHashedNodesCoordinator.go [hunk 1]
```diff
@@ -336,16 +334,13 @@ func (ihgs *indexHashedNodesCoordinator) GetValidatorWithPublicKey(
 		return nil, 0, ErrNilPubKey
 	}
 	ihgs.mutNodesConfig.RLock()
-	nodesConfig, ok := ihgs.nodesConfig[epoch]
-	ihgs.mutNodesConfig.RUnlock()
+	defer ihgs.mutNodesConfig.RUnlock()
```
Comment (SebastianMarian) [inline sharding/indexHashedNodesCoordinator.go:337]: Add before the return from line 284: ihgs.mutNodesConfig.RUnlock()

Comment (iulianpascalau) [reply sharding/indexHashedNodesCoordinator.go:339]: done

**Comment by AdoAdoAdo** [comment]
I suggest to close this PR as the original branch already solves the issue with the missing mutex unlock and the deadlock because of mutex in end of epoch trigger and channel from peer type provider.

**Comment by iulianpascalau** [comment]
Closed. The change will be postponed. This will be part of a larger refactoring task: https://elrondnetwork.atlassian.net/browse/EN-6445

**Pull request closed without merging**